sycl/include/sycl/ext/oneapi/experimental/bfloat16_math.hpp

```diff
@@ -232,6 +232,20 @@ sycl::marray<bfloat16, N> fma(sycl::marray<bfloat16, N> x,
   return res;
 }
 
+template <typename T>
+std::enable_if_t<std::is_same<T, bfloat16>::value, T> ceil(T x) {
+  return sycl::ext::oneapi::bfloat16{sycl::ceil(float{x})};
+}
+
+template <size_t N>
+sycl::marray<bfloat16, N> ceil(sycl::marray<bfloat16, N> x) {
+  sycl::marray<bfloat16, N> res;
+  for (size_t i = 0; i < N; i++) {
+    res[i] = ceil(x[i]);
+  }
+  return res;
+}
```

Comment:
Do we need to add this to [sycl_ext_oneapi_bfloat16_math_functions](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_bfloat16_math_functions.asciidoc), or is it defined by some other extension specification?

Reply:
Hi, @gmlueck 
I created the draft pr to ask for some feedback from your side, if you have no objection to add these generic impls based on fp32 emulation, I will add it in document too.
Thanks very much.

Reply:
Yes, I think it is OK for the generic implementation to use `float`.  Most of the operators on `bfloat16` work this way.  We convert the operands to `float`, perform the operation, then convert the result to `bfloat16`.  See the header "sycl/include/sycl/ext/oneapi/bfloat16.hpp".